apps/livreGUI/tfEditor/TransferFunctionEditor.cpp

```diff
@@ -178,43 +178,33 @@ void TransferFunctionEditor::_setDefault()
 
 void TransferFunctionEditor::_publishTransferFunction()
 {
-    if( ! _tfReceived )
-        return;
-
     const UInt8s& redCurve =  _redWidget->getCurve();
     const UInt8s& greenCurve =  _greenWidget->getCurve();
     const UInt8s& blueCurve =  _blueWidget->getCurve();
     const UInt8s& alphaCurve =  _alphaWidget->getCurve();
 
     if( redCurve.empty() || greenCurve.empty() || blueCurve.empty() ||
-        alphaCurve.empty( ))
+        alphaCurve.empty() || redCurve.size() * 4 != _lut.getLutSize( ))
     {
         return;
     }
 
-    UInt8s transferFunction;
-    transferFunction.reserve( redCurve.size() * 4u );
+    uint8_t* lut = _lut.getLut();
     for( uint32_t i = 0; i < redCurve.size(); ++i )
     {
-        transferFunction.push_back( redCurve[i] );
-        transferFunction.push_back( greenCurve[i] );
-        transferFunction.push_back( blueCurve[i] );
-        transferFunction.push_back( alphaCurve[i] );
+        lut[ 4*i + 0 ] = redCurve[i];
```

Comment:
Done what? It's commonly used in these situations - makes the code more consistent to read, and will be optimized out anyways.